test/end-to-end/Azure.Functions.NodejsWorker.E2E/Azure.Functions.NodejsWorker.E2E/HttpEndToEndTests.cs

```diff
@@ -25,7 +28,56 @@ public HttpEndToEndTests(FunctionAppFixture fixture)
         public async Task HttpTriggerTests(string functionName, string queryString, HttpStatusCode expectedStatusCode, string expectedMessage)
         {
             // TODO: Verify exception on 500 after https://github.com/Azure/azure-functions-host/issues/3589
-            Assert.True(await Utilities.InvokeHttpTrigger(functionName, queryString, expectedStatusCode, expectedMessage));
+            Assert.True(await Utilities.InvokeHttpTrigger(functionName, queryString, null, expectedStatusCode, expectedMessage));
+        }
+    }
+
+    [Collection(Constants.EasyAuthCollectionName)]
+    public class HttpEndToEndTests_Auth
+    {
+        private readonly EasyAuthFixture _fixture;
+
+        public HttpEndToEndTests_Auth(EasyAuthFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        [Fact]
+        public async Task HttpTrigger_Identities_Succeeds()
+        {
+            var headers = new Dictionary<string, string>
+            {
+                { "x-ms-client-principal", MockEasyAuth("facebook", "Connor McMahon", "10241897674253170") },
```

Comment:
This is just a question for my understanding, but having the setter on the request makes the host add a ClaimsPrincipal object on the functions-host side? So the stages of clearing auth are: "x-ms-client-principal" header is set on http reques, and iff identities are valid, they will appear as request.HttpContext.ClaimsPrincipal in C# which then gets passed as rpc object?

Reply:
And does all that claims info stay on the http header?

Reply:
Almost! Steps as I understand are:

1. If EasyAuth is enabled, the host's AuthenticationLevelHandler will get decode the base-64 encoded `x-ms-client-principal` header to a ClaimsPrincipal object.
2. ClaimsPrincipal object appears as request.HttpContext.User in C#.
3. In the host, ToRpc will translate the ClaimsPrincipal object to an RpcClaimsIdentity array on the RpcHttp object.
4. The language worker converts this to [end result language-dependent.]

The `x-ms-client-principal` header remains on the request. The ClaimsPrincipal binding is a convenience measure to keep header decoding boilerplate out of user code.